packages/ui-top-nav-bar/src/TopNavBar/props.ts

```diff
@@ -22,21 +22,48 @@
  * SOFTWARE.
  */
 
-import React from 'react'
 import PropTypes from 'prop-types'
 
+import { Children as ChildrenPropTypes } from '@instructure/ui-prop-types'
+
 import type { WithStyleProps, ComponentStyle } from '@instructure/emotion'
 import type {
   TopNavBarTheme,
   OtherHTMLAttributes,
   PropValidators
 } from '@instructure/shared-types'
 
+import { TopNavBarLayout } from './TopNavBarLayout'
+import { TopNavBarSmallViewportLayout } from './TopNavBarSmallViewportLayout'
+
+import type { ActionItemsChild } from './TopNavBarActionItems/props'
+import type { BrandChild } from './TopNavBarBrand/props'
+import type { ItemChild } from './TopNavBarItem/props'
+import type { LayoutChild } from './TopNavBarLayout/props'
+import type { MenuItemsChild } from './TopNavBarMenuItems/props'
+import type { SmallViewportLayoutChild } from './TopNavBarSmallViewportLayout/props'
+import type { UserChild } from './TopNavBarUser/props'
+
 type TopNavBarOwnProps = {
   /**
-   * FIXME: description of the children prop goes here
+   * TODO: description of the renderLayout prop goes here
+   */
+  renderLayout: LayoutChild
+
+  /**
+   * TODO: description of the renderSmallViewportLayout prop goes here
+   */
+  renderSmallViewportLayout?: SmallViewportLayoutChild
```

Comment:
I'd give these a more consistent name, say `renderLargeViewportLayout`, `renderSmallViewportLayout`.
Also the large layout could be optional too with a default value, like the small layout? I think its very handy if a component requires as little configuration as possible (=e.g. just setting text)